### PR TITLE
bump buildkit to 0.11.4

### PR DESCRIPTION
**Pull request by ndeloof**
**Related issue**
closes https://github.com/docker/compose/security/dependabot/21

**(not mandatory) A picture of a cute animal, if possible in relation to what you did**

### CODE REVIEW

**Review by milas — APPROVED** [review]

**Comment by glours** [comment]
I think we should use this [PR](https://github.com/docker/compose/pull/10794) instead to be sure the version of buildkit is align with the one defined in buildx

File: go.mod [hunk 1]
```diff
@@ -24,7 +24,7 @@ require (
 	github.com/jonboulle/clockwork v0.4.0
 	github.com/mattn/go-shellwords v1.0.12
 	github.com/mitchellh/mapstructure v1.5.0
-	github.com/moby/buildkit v0.11.0-rc3.0.20230609092854-67a08623b95a
+	github.com/moby/buildkit v0.11.4
```
Comment (thaJeztah) [inline go.mod:27]: ⚠️ v0.11.4 is **older** than `v0.11.0-rc3.0.20230609092854-67a08623b95a`. which is actually "v0.12-dev"


The `v0.11.0-rc3.0.20230609092854-67a08623b95a` pseudo version is based on the last tag that was cut from master (`v0.11.0-rc3`), which always makes it confusing. (sigh)

But if you look at the commit; https://github.com/moby/buildkit/commit/67a08623b95a, then you'll see it's a commit from master;
Screenshot 2023-07-17 at 01 51 03

**Review by thaJeztah — CHANGES_REQUESTED** [review]
see comment below

File: go.mod (hunk 1 above)
Comment (ndeloof) [reply go.mod:27]: > v0.11.4 is older than v0.11.0-rc3

seriously? 😰

Comment (thaJeztah) [reply go.mod:27]: Yeah, go modules tried to be smart and create versions for non-tagged commits, and it does so by;

- taking the commit
- find the last tag in the commit's history (in this case: master branch)
- if the tag cannot be parsed as SemVer, continue with the next one
- once a "SemVer" tag is found, take the tag and increment "Z" (`vX.Y.Z+1`)
- ...and that's what they use as "pseudo version" prefix


Which, erm, doesn't really work if a project uses release-branches, and the last tag was the pre-release before the release-branch was created.

**Pull request closed without merging**